pkg/util/error_list.go

```diff
@@ -45,10 +45,67 @@ func (list errorList) Error() string {
 	return result
 }
 
+// Except returns any errors (or nested errors, if the list contains
+// nested []error slices) for which all fns return false. If no errors
+// remain a nil list is returned. The resulting list will have all
+// nested []error slices flattened as a side effect.
```

Comment:
I think you're trying to position a very situation-specific solution in a generic way.  I accept that this might be the behavior you want, I just don't think it is useful outside this context.

And as for flattening, I *still* think that it's wrong.  You might want to recursively filter, but the result should still be a list of lists.  E.g.

given a list of errors `[ [A, B], [B], [B, C, D] ]`, and a desire to filter out B, the result should be `[ [A], [C, D] ]`, and not `[ A, C, D ]`

And now that I think about it, the act of flattening a slice to an error could probably do with an error message of its own (optional), so you'd get "Error frobnicating: [ [A], [C, D] ]"